aggsender/aggsender.go

```diff
@@ -263,8 +263,12 @@ func (a *AggSender) sendCertificates(ctx context.Context, returnAfterNIterations
 			iteration++
 			a.log.Debugf("Checking perodical certificates status (%s)",
 				a.cfg.CheckCertConfigBriefString())
-			checkResult := a.certStatusChecker.CheckPendingCertificatesStatus(ctx)
-			if !checkResult.ExistPendingCerts && checkResult.ExistNewInErrorCert {
+
+			checkResult, err := a.certStatusChecker.CheckPeriodicallyCertificateStatus(ctx)
+			if err != nil {
+				a.status.SetLastError(err)
+				a.log.Errorf("error checking last certificate from agglayer: %v", err)
+			} else if err == nil && !checkResult.ExistPendingCerts && checkResult.ExistNewInErrorCert {
```

Comment:
You don't need to add the check `if err == nil`, it is already nil if it gets to this `else if` since the condition in the `if` is `err != nil`.